src/CodeGeneration/ClassGenerator/JobGenerator.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Dealroadshow\Bundle\K8SBundle\CodeGeneration\ClassGenerator;
+
+use Dealroadshow\Bundle\K8SBundle\CodeGeneration\ClassDetails;
+use Dealroadshow\Bundle\K8SBundle\CodeGeneration\ClassDetailsResolver\ManifestResolver;
+use Dealroadshow\Bundle\K8SBundle\CodeGeneration\TemplateRenderer;
+use Dealroadshow\Bundle\K8SBundle\Util\Dir;
+use Dealroadshow\K8S\Framework\App\AppInterface;
+use Dealroadshow\K8S\Framework\Core\Job\JobInterface;
+use Dealroadshow\K8S\Framework\Registry\AppRegistry;
+use Dealroadshow\K8S\Framework\Registry\ManifestRegistry;
+use InvalidArgumentException;
+
+class JobGenerator implements ManifestGeneratorInterface
+{
+    private ManifestResolver $resolver;
+    private ManifestRegistry $manifestRegistry;
+    private AppRegistry $appRegistry;
+    private TemplateRenderer $renderer;
+
+    public function __construct(AppRegistry $appRegistry, ManifestRegistry $manifestRegistry, ManifestResolver $resolver, TemplateRenderer $renderer)
+    {
+        $this->resolver = $resolver;
+        $this->manifestRegistry = $manifestRegistry;
+        $this->appRegistry = $appRegistry;
+        $this->renderer = $renderer;
+    }
+
+    public function generate(string $appName, string $depName): string
```

Comment:
maybe you should rename second param to `$manifestName`, for it to be in sync with param in interface?